### PR TITLE
[GLT-3535] Added requisitions, assays, and metrics

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: docs/user_manual/type_data.md [hunk 1]
```diff
@@ -693,3 +693,31 @@ An attachment category can only be deleted if there are no attachments of that c
...
+the associated value.
+
+Examples:
+
+* A "Container Intact" metric for receipt requiring a pass/fail inspection of the received container
+* A "Yield" metric for library preparation requiring library yield to be greater than or equal to a threshold value in
+  ng/µL
+
+MISO administrators can add, edit, and delete metfics using the standard interface. A metric can only be deleted if it
+is not used by any existing studies.
```
Comment (callunity) [inline docs/user_manual/type_data.md:723]: `studies` -> `requisitions` or `projects`?

File: docs/user_manual/type_data.md [hunk 2]
```diff
@@ -693,3 +693,31 @@ An attachment category can only be deleted if there are no attachments of that c
...
+are specified in the assay, as different assays may have different thresholds. A metric may also specify the units for
+the associated value.
+
+Examples:
+
+* A "Container Intact" metric for receipt requiring a pass/fail inspection of the received container
+* A "Yield" metric for library preparation requiring library yield to be greater than or equal to a threshold value in
+  ng/µL
+
+MISO administrators can add, edit, and delete metfics using the standard interface. A metric can only be deleted if it
```
Comment (callunity) [inline docs/user_manual/type_data.md:722]: `metfics` -> `metrics`

File: docs/user_manual/requisitions.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+"Assays" link within. All of the existing assays are shown here, and the toolbar at the top of the list contains
+controls for working with assays.
+
+### Creating Assays
+
+To create a new assay, you must be a MISO administrator. Click the "Add" button in the toolbar at the top of the Assays
+list. This takes you to the Create Assay page. In the top section of the page, you can enter the basic information for
+the assay, including its alias, version, and description.
+
+Below, controls in the Metrics table allow you to define the assay's metrics. To add a metric, click the "Add" button.
```
Comment (callunity) [inline docs/user_manual/requisitions.md:83]: I've clicked the "Add" button and found that the "Add Metric" popup is entirely blank.

Comment (djcooke) [reply docs/user_manual/requisitions.md:83]: You have to create the metrics first before you can add them to an assay. I'll add a message to make that more obvious when there are no metrics though

Comment (callunity) [reply docs/user_manual/requisitions.md:83]: Ah, that makes sense :+1:

**Review by avarsava — APPROVED** [review]

**Review by callunity — APPROVED** [review]

**Comment by callunity** [comment]
Approved because frankly, the minor modifications to documentation I've suggested can even go into the next release.

File: docs/user_manual/requisitions.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+column. To create multiple samples in the same requisition at the same time, enter the same Requisition Alias and
+choose "Create New" in the Requisition column for all of them.
+
+### Create Requisition Page
+
+If you want to add a requisition to existing samples, you should use the Create Requistion page. To get there, go to
+the Requisitions list page and click the "Add" button in the toolbar.
+
+On the Create Requisition page, you can give the requisition an alias and choose an assay. The alias may refer to a
+requisition form stored in a separate system, or may simply be a friendly identifier for the requisition. You must save
```
[the remark this answers is not in the document]
Comment (djcooke) [reply docs/user_manual/requisitions.md]: fixed

File: docs/user_manual/requisitions.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+Be careful when editing assays, as this will affect any requisitions that the assay is already linked to. If the assay
+has changed, it is probably best to create a new assay with the same alias and a different version. That way, you
+retain the requirements that were used for existing requisitions, while updating the requirements going forward. Assays
+may be archived, so you can archive the older version in order to prevent it from being used in the future.
+
+If you do wish to edit an assay, click its alias on the Assays list. This will take you to the View Assay page. You must
+be a MISO administrator to edit the assay. If you are an administrator, you can click the Edit button at the top right
+and confirm that you want to edit. This takes you to the Edit Assay page, where you can make changes. The version is
+locked, as new assays should be created to represent updated versions. Any other changes can be made similarly to on
```
[the remark this answers is not in the document]
Comment (djcooke) [reply docs/user_manual/requisitions.md:100]: That would mean the changes could be made on the Create Assay page. They can be made on the Edit Assay page, in a way that is similar to the Create Assay page.

File: docs/user_manual/type_data.md (hunk 1 above)
Comment (djcooke) [reply docs/user_manual/type_data.md]: Changed to "assays."

File: docs/user_manual/type_data.md (hunk 2 above)
Comment (djcooke) [reply docs/user_manual/type_data.md]: Fixed

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/qc/RequisitionQC.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2012. The Genome Analysis Centre, Norwich, UK
+ * MISO project contacts: Robert Davey @ TGAC
```
[the remark this answers is not in the document]
Comment (djcooke) [reply core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/qc/RequisitionQC.java]: New files usually don't have the block at all. Removed

**Pull request merged**